### ISSUE TITLE
shouldn't IArrakisV2 be explicitely implemented by ArrakisV2 vault

### ISSUE THREAD

**Issue by ferencdg** [ArrakisFinance/v2-core#170]
Currently I don't see any 'is IArrakis' when defining the ArrakisV2 valult. I was doing some refactoring on my fork, and renamed one of the methods inside the ArrakisV2 vault (getManagerBps), which was also defined in the IArrakisVault2. Sadly the contract compiled and then I got a runtime error after deployment for the following codeline:

```solidity
 IArrakisV2 arrakisV2 = IArrakisV2(underlyingPayload_.self);

        (uint256 fee0After, uint256 fee1After) = subtractAdminFees(
            fee0,
            fee1,
            arrakisV2.managerFeeBPS()
        );
```

**Comment by kassandraoftroy** [comment]
Yeah it's a good point (if the ArrakisV2 contract were to implement IArrakisV2 explicitly this would be a compile time error rather than runtime)
